### PR TITLE
:sparkles: A little refactor

### PR DESCRIPTION
**Pull request by FurryR**
# Update

- ja-JP, zh-CN l10n
- Brand-new universal `getBlocklyInstance` method (removed unnecessary alternative method)
- Prettier support

### CODE REVIEW

File: .prettierrc.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "semi": true,
+  "tabWidth": 4,
+  "singleQuote": true,
+  "trailingComma": "es5",
+  "arrowParens": "always",
+  "bracketSameLine": true,
+  "bracketSpacing": true,
+  "printWidth": 80
+}
```
Comment (SimonShiki) [inline .prettierrc.json]: add new line

File: src/frontend/index.ts [hunk 2]
```diff
@@ -19,49 +19,81 @@ interface MothDispatchedAllocate {
 
 type MothDispatched = MothDispatchedAllocate | MothDispatchedLoad;
 
-function getExtensionInfo () {
+/**
+ * Get all extensions.
+ * @returns Extensions.
+ */
+function getExtensionInfo() {
```
Comment (SimonShiki) [inline src/frontend/index.ts:26]: space between function name and brackets. please edit prettier config

File: src/frontend/index.ts [hunk 3]
```diff
@@ -19,49 +19,81 @@ interface MothDispatchedAllocate {
...
     const processedExtInfo: MothExtensionInfo[] = [];
-    for (const [extId, ext] of window.chibi.loader.loadedScratchExtension.entries()) {
+    for (const [
+        extId,
+        ext,
+    ] of window.chibi.loader.loadedScratchExtension.entries()) {
```
Comment (SimonShiki) [inline src/frontend/index.ts]: unnecessary line breaks. please edit prettier config

File: src/frontend/index.ts [hunk 4]
```diff
@@ -19,49 +19,81 @@ interface MothDispatchedAllocate {
...
         processedExtInfo.push({
             name: extId,
-            sandboxed: ext.env === 'sandboxed'
+            sandboxed: ext.env === 'sandboxed',
```
Comment (SimonShiki) [inline src/frontend/index.ts]: unnecessary comma. please edit prettier config

File: src/frontend/index.ts [hunk 5]
```diff
@@ -19,49 +19,81 @@ interface MothDispatchedAllocate {
...
-        break;
-    case 'load':
-        await window.chibi.loader.load(event.data.info.url, event.data.info.sandboxed ? 'sandboxed' : 'unsandboxed');
-        dashboardWindow?.postMessage({
-            type: 'extension',
-            extensions: getExtensionInfo()
-        }, '*');
-        break;
+        case 'allocate':
+            // Handshake: send current extension info in order to prepare frontend.
```
Comment (SimonShiki) [inline src/frontend/index.ts]: put it on the previous line

File: src/frontend/index.ts [hunk 6]
```diff
@@ -19,49 +19,81 @@ interface MothDispatchedAllocate {
...
+            dashboardWindow?.postMessage(
+                {
+                    type: 'extension',
+                    extensions: getExtensionInfo(),
+                },
+                '*'
+            );
+            break;
+        case 'load':
+            // Load a extension.
```
Comment (SimonShiki) [inline src/frontend/index.ts]: put it on the previous line
also, use 'an'

File: src/frontend/index.ts [hunk 7]
```diff
@@ -19,49 +19,81 @@ interface MothDispatchedAllocate {
...
+ * @param open window.open function (compatible with ccw).
+ */
+function openFrontend(open: typeof window.open) {
+    dashboardWindow = open(
+        'https://chibi.codingclip.cc/#manage',
+        'Chibi',
+        'popup=yes,status=no,location=no,toolbar=no,menubar=no'
+    );
 }
-
```
Comment (SimonShiki) [inline src/frontend/index.ts:66]: add new line. please edit prettier config if necessary

File: src/global.d.ts [hunk 8]
```diff
@@ -8,10 +8,13 @@ declare interface Window {
         vm?: VM;
         blockly?: ScratchBlocks | null;
         loader?: ChibiLoader;
-        registeredExtension: Record<string, {
```
Comment (SimonShiki) [inline src/global.d.ts:11]: unnecessary line breaks. please edit prettier config

File: src/injector/inject.ts [hunk 9]
```diff
@@ -1,57 +1,79 @@
...
+            // it is an array, hijack every listeners.
+            for (const value of events) {
+                const result = hijack(value);
+                if (result && result.ScratchBlocks) return result.ScratchBlocks;
+            }
+        }
+    }
+    return undefined; // Method failed.
 }
-
```
Comment (SimonShiki) [inline src/injector/inject.ts:43]: new line requested.
add new line between functions. please edit prettier config

File: src/injector/inject.ts [hunk 10]
```diff
@@ -1,57 +1,79 @@
...
+ * @param open window.open function (compatible with ccw).
+ * @return Callback promise. After that you could use window.chibi.vm to get the virtual machine.
+ */
+export function trap(open: typeof window.open): Promise<void> {
     window.chibi = {
         // @ts-expect-error defined in webpack define plugin
         version: __CHIBI_VERSION__,
         registeredExtension: {},
-        openFrontend: openFrontend
+        openFrontend: openFrontend.bind(null, open),
```
Comment (SimonShiki) [inline src/injector/inject.ts]: 😎

File: src/injector/inject.ts [hunk 11]
```diff
@@ -63,8 +85,11 @@ export function trap () {
                 return oldBind.apply(this, args);
             } else if (
                 args[0] &&
-                Object.prototype.hasOwnProperty.call(args[0], "editingTarget") &&
-                Object.prototype.hasOwnProperty.call(args[0], "runtime")
+                Object.prototype.hasOwnProperty.call(
+                    args[0],
```
Comment (SimonShiki) [inline src/injector/inject.ts]: unnecessary line breaks!!!!!!

File: src/injector/inject.ts [hunk 12]
```diff
@@ -139,35 +201,48 @@ export function inject (vm: ChibiCompatibleVM) {
...
+        args: Record<string, unknown>,
+        ...otherArgs: unknown[]
+    ) {
         const chibiFlag = args.VALUE;
         switch (chibiFlag) {
-        case '🧐 Chibi Installed?':
-            return true;
-        default:
-            return originalArgReporterBooleanFunc.call(this, args, ...otherArgs);
+            case '🧐 Chibi?':
```
Comment (SimonShiki) [inline src/injector/inject.ts:227]: I'm concerned that this change will break compatibility with previous versions...

File: src/injector/inject.ts [hunk 13]
```diff
@@ -139,35 +201,48 @@ export function inject (vm: ChibiCompatibleVM) {
...
+            ...args: unknown[]
+        ) {
             for (const extensionId of extensions) {
                 if (
                     !vm.ccExtensionManager!.info.hasOwnProperty(extensionId) &&
                     extensionId in window.chibi.registeredExtension
                 ) {
                     vm.ccExtensionManager!.info[extensionId] = {
-                        api: 0
+                        api: 0,
```
Comment (SimonShiki) [inline src/injector/inject.ts]: unnecessary comma

File: src/injector/inject.ts [hunk 14]
```diff
@@ -178,67 +253,7 @@ export function inject (vm: ChibiCompatibleVM) {
 
     // Blockly stuffs
     setTimeout(() => {
-        const blockly = window.chibi.blockly = getBlocklyInstance();
```
Comment (SimonShiki) [inline src/injector/inject.ts:181]: the new ways to get Blockly look exciting. but I still want to keep this alternative method to maintain compatibility as much as possible.

File: src/loader/dispatch/central-dispatch.ts [hunk 15]
```diff
@@ -12,14 +12,15 @@ class _CentralDispatch extends SharedDispatch {
...
+    workerClass: typeof Worker | null =
+        typeof Worker === 'undefined' ? null : Worker;
     /**
      * List of workers attached to this dispatcher.
      * @type {Array}
      */
     workers: Worker[] = [];
     _onMessage!: (worker: Worker, event: MessageEvent) => void;
-    constructor () {
+    constructor() {
```
Comment (SimonShiki) [inline src/loader/dispatch/central-dispatch.ts:22]: space between function name and brackets

File: src/loader/dispatch/central-dispatch.ts [hunk 16]
```diff
@@ -114,18 +127,20 @@ class _CentralDispatch extends SharedDispatch {
...
      * @protected
      */
-    _onDispatchMessage (worker: Worker, message: DispatchCallMessage) {
+    _onDispatchMessage(worker: Worker, message: DispatchCallMessage) {
         let promise;
         switch (message.method) {
-        case 'setService':
-            if (!message.args) {
-                console.error('setService received empty argument');
+            case 'setService':
```
Comment (SimonShiki) [inline src/loader/dispatch/central-dispatch.ts:124]: 😎

File: src/loader/dispatch/worker-dispatch.ts [hunk 17]
```diff
@@ -77,12 +78,12 @@ class _WorkerDispatch extends SharedDispatch {
...
      * @protected
      */
-    _getServiceProvider (service: string) {
+    _getServiceProvider(service: string) {
         // If we don't have a local service by this name, contact central dispatch by calling `postMessage` on self
         const provider = this.services[service];
         return {
             provider: provider || self,
-            isRemote: !provider
+            isRemote: !provider,
```
Comment (SimonShiki) [inline src/loader/dispatch/worker-dispatch.ts]: unnecessary comma

File: src/typings/block-type.ts [hunk 18]
```diff
@@ -41,9 +41,7 @@ enum BlockType {
     /**
      * Lambda reporter with callable function.
```
Comment (SimonShiki) [inline src/typings/block-type.ts]: this is a private type from my own scratch mod. I recommend removing it.
sorry for causing this trouble...

**Review by SimonShiki — CHANGES_REQUESTED** [review]
- the code style problem introduced by prettier is absolutely a disaster... I will specifically note all the places where I need to change the prettier configuration.
- the l10n part can be replaced by ``formatMessage``. Using standardized solutions can pave the way for subsequent i18n.

anyway, some of the changes in this pull request are very exciting. thank you very much for your contribution to this project!

File: src/loader/loader.ts [hunk 19]
```diff
@@ -233,34 +265,53 @@ class ChibiLoader {
...
+                                serviceName
+                            );
+                            break;
+                    }
+                    results.push(result);
+                } catch (e: unknown) {
+                    // TODO: more meaningful error reporting
+                    error(
+                        `Error processing block: ${
+                            (e as Error).message
```
Comment (SimonShiki) [inline src/loader/loader.ts]: UNNECESSARY LINE BREAKS!!!!!

File: src/injector/inject.ts (hunk 11 above)
Comment (FurryR) [reply src/injector/inject.ts]: Prettier says it is necessary.

File: src/injector/inject.ts (hunk 10 above)
Comment (FurryR) [reply src/injector/inject.ts]: Simple curry function to ensure compatibility.

File: src/injector/inject.ts (hunk 12 above)
Comment (FurryR) [reply src/injector/inject.ts:227]: Yes it will. I will fix this later.

Comment (FurryR) [reply src/injector/inject.ts:227]: Since the block cannot be translated, I used a universal name.

File: src/injector/inject.ts (hunk 13 above)
Comment (FurryR) [reply src/injector/inject.ts]: There are no extra commas.

File: .prettierrc.json (hunk 1 above)
Comment (FurryR) [reply .prettierrc.json]: Added

File: src/frontend/index.ts (hunk 2 above)
Comment (FurryR) [reply src/frontend/index.ts:26]: Conflict: prettier does not support this feature.

File: src/injector/inject.ts (hunk 14 above)
Comment (FurryR) [reply src/injector/inject.ts:181]: Okay.

File: src/injector/inject.ts (hunk 9 above)
Comment (FurryR) [reply src/injector/inject.ts:43]: Done.

File: src/global.d.ts (hunk 8 above)
Comment (FurryR) [reply src/global.d.ts:11]: It is valid.

File: src/loader/dispatch/central-dispatch.ts (hunk 16 above)
Comment (FurryR) [reply src/loader/dispatch/central-dispatch.ts:124]: What?

File: src/loader/dispatch/worker-dispatch.ts (hunk 17 above)
Comment (FurryR) [reply src/loader/dispatch/worker-dispatch.ts]: Fixed

File: src/loader/loader.ts (hunk 19 above)
Comment (FurryR) [reply src/loader/loader.ts]: Fixed

File: src/typings/block-type.ts (hunk 18 above)
Comment (FurryR) [reply src/typings/block-type.ts]: No, it is necessary. See `make-ctx`.ts.

File: src/loader/dispatch/central-dispatch.ts (hunk 16 above)
Comment (SimonShiki) [reply src/loader/dispatch/central-dispatch.ts:124]: compliments for this change

File: src/injector/inject.ts (hunk 11 above)
Comment (SimonShiki) [reply src/injector/inject.ts]: It's unnecessary. It’s neither pretty nor does it improve readability much

File: src/global.d.ts (hunk 8 above)
Comment (SimonShiki) [reply src/global.d.ts:11]: it’s neither pretty nor does it improve readability much

File: src/frontend/index.ts (hunk 2 above)
Comment (SimonShiki) [reply src/frontend/index.ts:26]: 😨 if it's caused by prettier and unable to disable. please consider working with eslint or removing prettier.

Comment (FurryR) [reply src/frontend/index.ts:26]: > 😨 if it's caused by prettier and unable to disable. please consider working with eslint or removing prettier.

Eslint is not working either! It cannot fix errors like this:

```js
if (condition) {
};
```

`;` is unnecessary.

**Review by SimonShiki — APPROVED** [review]

**Pull request merged**